File: miniclaw/config/paths.py
"""Path utilities for Miniclaw."""
from pathlib import Path

def get_app_dir() -> Path:
    path = Path.home() / ".miniclaw"
    path.mkdir(parents=True, exist_ok=True)
    return path

def get_config_path() -> Path:
    return get_app_dir() / "config.json"

def get_env_path() -> Path:
    return get_app_dir() / ".env"

def get_workspace_dir(workspace_path: str = None) -> Path:
    path = Path(workspace_path).expanduser() if workspace_path else get_app_dir() / "workspace"
    path.mkdir(parents=True, exist_ok=True)
    return path

File: src/config/paths.ts
import fs from "node:fs";
import os from "node:os";
import path from "node:path";

export function getAppDir(): string {
	const dir = path.join(os.homedir(), ".miniclaw");
	if (!fs.existsSync(dir))
		fs.mkdirSync(dir, {
			recursive: true,
		});
	return dir;
}

export function getConfigPath(): string {
	return path.join(getAppDir(), "config.json");
}

export function getEnvPath(): string {
	return path.join(getAppDir(), ".env");
}

export function getWorkspaceDir(customPath?: string): string {
	const wsPath = customPath
		? customPath.replace("~", os.homedir())
		: path.join(getAppDir(), "workspace");
	if (!fs.existsSync(wsPath))
		fs.mkdirSync(wsPath, {
			recursive: true,
		});
	return wsPath;
}
